### PR TITLE
Update SoundTouch to latest

### PR DESCRIPTION
**Pull request by Filoppi**
Based on https://gitlab.com/soundtouch/soundtouch/-/commit/3e74d1d18f35ab127c9702b7ccda5bfb093d9219
Unfortunately the library wasn't vanilla as it moved files and includes around, but the only actual code changes by Dolphin were a few defines which were enabled or disabled in STTypes.h.

### CODE REVIEW

File: Externals/soundtouch/PeakFinder.cpp [hunk 1]
```diff
@@ -64,7 +57,7 @@ int PeakFinder::findTop(const float *data, int peakpos) const
 
     refvalue = data[peakpos];
 
-    // seek within �10 points
+    // seek within �10 points
```
Comment (Rumi-Larry) [inline Externals/soundtouch/PeakFinder.cpp:60]: This character wasn't correctly interpreted by Git. It shows a replacement character.

File: Externals/soundtouch/PeakFinder.cpp [hunk 2]
```diff
@@ -265,7 +256,7 @@ double PeakFinder::detectPeak(const float *data, int aminPos, int amaxPos)
 
         // accept harmonic peak if 
         // (a) it is found
-        // (b) is within �4% of the expected harmonic interval
+        // (b) is within �4% of the expected harmonic interval
```
Comment (Rumi-Larry) [inline Externals/soundtouch/PeakFinder.cpp:259]: ditto

File: Externals/soundtouch/SoundTouch.cpp [hunk 3]
```diff
@@ -520,7 +523,16 @@ uint SoundTouch::receiveSamples(SAMPLETYPE *output, uint maxSamples)
...
 {
-	uint ret = FIFOProcessor::receiveSamples(maxSamples);
-	samplesOutput += (long)ret;
-	return ret;
+    uint ret = FIFOProcessor::receiveSamples(maxSamples);
+    samplesOutput += (long)ret;
+    return ret;
+}
+
+
```
Comment (Rumi-Larry) [inline Externals/soundtouch/SoundTouch.cpp:531]: Unnecessary extra blank line?

File: Externals/soundtouch/SoundTouch.h [hunk 4]
```diff
@@ -116,30 +109,61 @@ namespace soundtouch
...
 /// - This is read-only parameter, i.e. setSetting ignores this parameter
-/// - Returned value is approximate average value, exact processing batch
-///   size may wary from time to time
-/// - This parameter value is not constant but may change depending on 
+/// - This parameter value is not constant but change depending on 
 ///   tempo/pitch/rate/samplerate settings.
-#define SETTING_NOMINAL_OUTPUT_SEQUENCE		7
+#define SETTING_NOMINAL_OUTPUT_SEQUENCE     7
+
+
```
Comment (Rumi-Larry) [inline Externals/soundtouch/SoundTouch.h:137]: Ditto. I only mention it for consistency.

File: Externals/soundtouch/PeakFinder.cpp (hunk 1 above)
Comment (Filoppi) [reply Externals/soundtouch/PeakFinder.cpp:60]: https://gitlab.com/soundtouch/soundtouch/-/blob/master/source/SoundTouch/PeakFinder.cpp
It's the same in the official repository.

File: Externals/soundtouch/SoundTouch.cpp (hunk 3 above)
Comment (Filoppi) [reply Externals/soundtouch/SoundTouch.cpp:531]: Not sure what you'd want me to do, this is the library as it is, unchanged. A lot of typos and things like this had been fixed, while some new ones introduced apparently. I don't think it's worth fixing, and I was told that dolphin usually doesn't change externals.

Comment (Filoppi) [reply Externals/soundtouch/SoundTouch.cpp:531]: They actually added it for consistency, as they always have 2 spaces between methods:
https://gitlab.com/soundtouch/soundtouch/-/blob/master/source/SoundTouch/SoundTouch.cpp

File: Externals/soundtouch/SoundTouch.h (hunk 4 above)
Comment (Filoppi) [reply Externals/soundtouch/SoundTouch.h:137]: They actually added it for consistency, as they always have 2 spaces between defines:
https://gitlab.com/soundtouch/soundtouch/-/blob/master/include/SoundTouch.h

**Comment by Tilka** [comment]
Can you squash the commits please?

**Comment by Filoppi** [comment]
> Can you squash the commits please?

Done

**Comment by Mystro256** [comment]
Does this fix the blocking windows build issue in https://github.com/dolphin-emu/dolphin/pull/8725?

**Comment by Filoppi** [comment]
> Does this fix the blocking windows build issue in #8725?

I think this was building fine.

**Comment by Mystro256** [comment]
Ok, well feel free to use my pull request if a stable release version of sound touch is desired.

**Comment by Filoppi** [comment]
> Ok, well feel free to use my pull request if a stable release version of sound touch is desired.

This was stable and more up to date, I've been testing it for a few months.

**Comment by PatrickFerry** [comment]
This has been superseded by -> https://github.com/dolphin-emu/dolphin/pull/11706

**Pull request closed without merging**